tools/pythonpkg/tests/fast/test_runtime_error.py

```diff
@@ -116,7 +116,10 @@ def test_conn_broken_statement_error(self, pandas):
     def test_conn_prepared_statement_error(self):
         conn = duckdb.connect()
         conn.execute("create table integers (a integer, b integer)")
-        with pytest.raises(duckdb.InvalidInputException, match='Prepared statement needs 2 parameters, 1 given'):
+        with pytest.raises(
+            duckdb.InvalidInputException,
+            match='Values were not provided for the following prepared statement parameters: 2',
```

Comment:
Does this error message need to change?

Reply:
I think so: Before this PR, this error message was generated in `tools/pythonpkg/src/pyconnection.cpp` by `TransformPreparedParameters(PreparedStatement &prep, const py::object &params)` However, now we are passing both the query and the parameters to PendingQuery in one go. 

I could try to refactor the error message code in main duckdb here to match this, but then i will just end up having to change the error message in other places probably

Reply:
Makes sense, thanks!